### PR TITLE
enh(api/token): Added new API endpoint to list authentication tokens

### PR DESCRIPTION
**Pull request by callapa**
## Description

Added new API endpoint to list authentication tokens.
```
GET /administration/tokens
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+get:
+  tags:
+    - Administration
+    - Authentication token
+  summary: "Find all Api authentication tokens"
+  description: |
+    Return all Api authentication token.
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml:7]:
```suggestion
    Return all API authentication tokens.
```

File: centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+get:
+  tags:
+    - Administration
+    - Authentication token
+  summary: "Find all Api authentication tokens"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml:5]:
```suggestion
  summary: "Find all API authentication tokens"
```

File: centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                $ref: '../../Common/Schema/Meta.yaml'
+    '403':
+      $ref: '../../Common/Response/Forbidden.yaml'
+    '500':
+      $ref: '../../Common/Response/InternalServerError.yaml'
+post:
+  tags:
+    - Administration
+    - Authentication token
+  summary: "Add an Api authentication token"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml:46]:
```suggestion
  summary: "Add an API authentication token"
```

File: centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  description: "Add an Api authentication token"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/AddTokenAndFindTokens.yaml:47]:
```suggestion
  description: "Add an API authentication token"
```

File: centreon/doc/API/v23.10/Administration/Token/Schema/NewToken.yaml [hunk 5]
```diff
@@ -2,14 +2,14 @@ type: object
 properties:
   name:
     type: string
-    description: "Token name."
+    description: "Token name"
     example: "my-api-token"
   user_id:
     type: integer
-    description: "User ID linked to the token"
+    description: "User ID linked to the Api authentication token"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/Schema/NewToken.yaml:9]:
```suggestion
    description: "User ID linked to the API authentication token"
```

File: centreon/doc/API/v23.10/Administration/Token/Schema/NewToken.yaml [hunk 6]
```diff
@@ -2,14 +2,14 @@ type: object
...
     example: 23
   expiration_date:
     type: string
-    description: "Expiration date of the token (date format should be in ISO 8601)"
+    description: "Expiration date of the Api authentication token"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/Schema/NewToken.yaml:13]:
```suggestion
    description: "Expiration date of the API authentication token"
```

File: centreon/doc/API/v23.10/Administration/Token/Schema/Token.yaml [hunk 7]
```diff
@@ -29,19 +29,15 @@ properties:
         example: "Jane Doe"
   creation_date:
     type: string
-    description: "Creation date of the token (date format should be in ISO 8601)"
+    description: "Creation date of the Api authentication token"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Administration/Token/Schema/Token.yaml:32]:
```suggestion
    description: "Creation date of the API authentication token"
```

File: centreon/src/Core/Security/Token/Infrastructure/Repository/DbReadTokenRepository.php [hunk 8]
```diff
@@ -119,4 +129,117 @@ public function existsByNameAndUserId(string $tokenName, int $userId): bool
...
+            INNER JOIN `:db`.security_token provider_token
+                ON provider_token.id = sat.provider_token_id
+            INNER JOIN `:db`.contact
+                ON contact.contact_id = sat.user_id
+            SQL;
+
+        // Search
+        $search = $sqlRequestTranslator->translateSearchParameterToSql();
+        $search .= $search === null ? ' WHERE ' : ' AND ';
+        $search .= 'sat.token_type = \'manual\'';
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Infrastructure/Repository/DbReadTokenRepository.php:175]: Use constant for manual ?

**Review by a-launois — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**